beacon_node/http_api/src/lib.rs

```diff
@@ -736,16 +769,24 @@ pub fn serve<T: BeaconChainTypes>(
         }))
         .and(warp::path::end())
         .and_then(
-            |state_id: StateId, chain: Arc<BeaconChain<T>>, validator_id: ValidatorId| {
-                blocking_json_task(move || {
+            |state_id: StateId,
+             task_spawner: TaskSpawner<T::EthSpec>,
+             chain: Arc<BeaconChain<T>>,
+             validator_id: ValidatorId| {
+                task_spawner.blocking_json_task(Priority::P1, move || {
                     let (data, execution_optimistic, finalized) = state_id
                         .map_state_and_execution_optimistic_and_finalized(
                             &chain,
                             |state, execution_optimistic, finalized| {
                                 let index_opt = match &validator_id {
-                                    ValidatorId::PublicKey(pubkey) => {
-                                        state.validators().iter().position(|v| v.pubkey == *pubkey)
-                                    }
+                                    ValidatorId::PublicKey(pubkey) => pubkey_to_validator_index(
```

Comment:
This is the only route where I've changed business logic, all the others should be the same! Unfortunately, cargo-fmt has caused a pretty noisy diff for the others.

See https://github.com/sigp/lighthouse/pull/4462/commits/efbabe32521ed6eb3564764da4e507d26a1c4bd0 for the change to this endpoint.